File: app/src/main/java/com/elvotra/clean/data/local/model/PostEntity.java
package com.elvotra.clean.data.local.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(tableName = "posts",
        primaryKeys = {"posts_id"},
        foreignKeys = @ForeignKey(entity = UserEntity.class,
            parentColumns = "users_id",
            childColumns = "user_id",
            onDelete=CASCADE))
public class PostEntity {

    @ColumnInfo(name = "posts_id")
    private int id;

    @ColumnInfo(name = "user_id")
    private int userId;

    @ColumnInfo(name = "title")
    private String title;

    @ColumnInfo(name = "body")
    private String body;

    public PostEntity(int id, int userId, String title, String body) {
        this.id = id;
        this.userId = userId;
        this.title = title;
        this.body = body;
    }

    public int getId() {
        return id;
    }

    public int getUserId() {
        return userId;
    }

    public String getTitle() {
        return title;
    }

    public String getBody() {
        return body;
    }
}

File: app/src/main/java/com/elvotra/clean/data/local/model/PostEntity.kt
package com.elvotra.clean.data.local.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

@Entity(tableName = "posts")
class PostEntity(@field:PrimaryKey
                 @field:ColumnInfo(name = "posts_id")
                 val id: Int, @field:ColumnInfo(name = "user_id")
                 val userId: Int, @field:ColumnInfo(name = "title")
                 val title: String, @field:ColumnInfo(name = "body")
                 val body: String) {

    override fun toString(): String {
        return "PostEntity{" +
                "id=" + id +
                ", userId=" + userId +
                ", title='" + title + '\''.toString() +
                ", body='" + body + '\''.toString() +
                '}'.toString()
    }
}
